File: Java/StrategyPattern/src/com/laxmena/VideoStreamerClient.java
package com.laxmena;

public class VideoStreamerClient {

    public void play(VideoChunk chunk) {
        System.out.println("Playing: " + chunk.getVideoChunkContent());
    }

    public static void main(String[] args) {
	    // Create a new VideoStreamer
        VideoStreamerClient client = new VideoStreamerClient();

        VideoStreamer streamer = new VideoStreamer();
        client.play(streamer.getVideoChunk());
        client.play(streamer.getVideoChunk());

        System.out.println("Network Speed update: 40");
        streamer.setNetworkSpeed(40);
        client.play(streamer.getVideoChunk());
        client.play(streamer.getVideoChunk());

        System.out.println("Network Speed update: 120");
        streamer.setNetworkSpeed(120);
        client.play(streamer.getVideoChunk());
        client.play(streamer.getVideoChunk());

        System.out.println("Network Speed update: 80");
        streamer.setNetworkSpeed(80);
        client.play(streamer.getVideoChunk());
        client.play(streamer.getVideoChunk());

    }
}

File: Python1/StrategyPattern/videostreamer_client.py
from VideoChunk.video_chunk import VideoChunk
from videostreamer import VideoStreamer

def play_video(chunk: VideoChunk):
    print("Now Playing: Chunk ID - ", chunk.get_chunk_id(), " ==> ", chunk.get_content())

if __name__ == '__main__':
    network_speed = 70
    # Create Streamer object
    streamer = VideoStreamer(network_speed)

    play_video(streamer.get_video_chunk())
    play_video(streamer.get_video_chunk())

    print("update network speed to 100")
    network_speed = 100
    streamer.set_network_speed(network_speed)
    play_video(streamer.get_video_chunk())
    play_video(streamer.get_video_chunk())
    play_video(streamer.get_video_chunk())

    print("update network speed to 20")
    network_speed = 20
    streamer.set_network_speed(network_speed)
    play_video(streamer.get_video_chunk())
    play_video(streamer.get_video_chunk())



